config/src/main/java/org/springframework/security/config/annotation/web/configurers/RequestCacheConfigurer.java

```diff
@@ -120,28 +120,19 @@ private RequestCache getRequestCache(H http) {
 		if (result != null) {
 			return result;
 		}
-		result = getBeanOrNull(RequestCache.class);
-		if (result != null) {
-			return result;
+
+		var context = getBuilder().getSharedObject(ApplicationContext.class);
+		if (context != null) {
+			result = context.getBeanProvider(RequestCache.class).getIfUnique();
+			if (result != null) {
+				return result;
+			}
 		}
 		HttpSessionRequestCache defaultCache = new HttpSessionRequestCache();
 		defaultCache.setRequestMatcher(createDefaultSavedRequestMatcher(http));
 		return defaultCache;
 	}
 
-	private <T> T getBeanOrNull(Class<T> type) {
-		ApplicationContext context = getBuilder().getSharedObject(ApplicationContext.class);
-		if (context == null) {
-			return null;
-		}
-		try {
-			return context.getBean(type);
```

Comment:
Same here, it will error if there is more than one bean, so let's use `getIfAvailable`.

Reply:
It's also same `getIfUnique` comment